File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/response/GetHintsResponse.java
package com.github.ai.leetcodequiz.api.response;

import com.github.ai.leetcodequiz.api.QuestionItemDto;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record GetHintsResponse(
    @NotNull List<QuestionItemDto> questions
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/response/GetHintsResponse.scala
package com.github.ai.leetcodequiz.api.response

import com.github.ai.leetcodequiz.api.QuestionItemDto
import zio.json.{JsonDecoder, JsonEncoder}

case class GetHintsResponse(
  questions: List[QuestionItemDto]
) derives JsonEncoder,
      JsonDecoder
